OrbitQt/CallTreeWidget.cpp

```diff
@@ -103,6 +99,42 @@ void CallTreeWidget::SetBottomUpView(std::unique_ptr<CallTreeView> bottom_up_vie
   ui_->callTreeTreeView->hideColumn(CallTreeViewItemModel::kExclusive);
 }
 
+void CallTreeWidget::resizeEvent(QResizeEvent* /*event*/) {
+  if (column_resizing_state_ == ColumnResizingState::kInitial) {
+    column_resizing_state_ = ColumnResizingState::kWidgetSizeSet;
+    ResizeColumnsIfNecessary();
+  }
+}
+
+void CallTreeWidget::ResizeColumnsIfNecessary() {
+  // Resize columns only once, when the following holds:
+  // - a non-empty CallTreeView is set;
+  // - the size of the QTreeView has been set, which means that the size of CallTreeWidget has been
+  //   set, which in turns means that the tab has been shown at least once.
+  if (column_resizing_state_ != ColumnResizingState::kWidgetSizeSet ||
+      hooked_proxy_model_ == nullptr || hooked_proxy_model_->rowCount(QModelIndex{}) == 0) {
+    return;
+  }
+
+  ui_->callTreeTreeView->header()->resizeSections(QHeaderView::ResizeToContents);
+
+  // Shrink the first column (to a minimum width) so that all columns are visible,
+  // but still make it as wide as possible.
+  static constexpr int kMinThreadOrFunctionColumnSize = 200;
+  int thread_or_function_column_size = ui_->callTreeTreeView->header()->width();
```

Comment:
I tested this only on Linux and maybe it works differently on Windows, but this `ui_->callTreeTreeView->header()->width()` does not return the maximum size the header can be (without a scrollbar being necessary), but some smaller size. The result is that the last column is taking up a lot of space and not the first. 

Maybe I am missing something, but intuitively I would say the first column in this screenshot should be the big one. 
![Screenshot 2020-10-27 at 16 34 12](https://user-images.githubusercontent.com/7501835/97324588-49e4c580-1872-11eb-8c82-b74d0f0047e6.png)

Reply:
In theory this is correct because the first column doesn't need shrinking. Maybe we also want to expand to take all the available space? 

So `ui_->callTreeTreeView->header()->width()` is less that `ui_->callTreeTreeView->width()`? I'll have a look.

Reply:
So this is because when more horizontal space is available than needed, `resizeSections(QHeaderView::ResizeToContents)` gives all the remaining space to the last column.
This is avoided by temporarily setting `setStretchLastSection(false)` before performing the computations.
Please have a look.

Reply:
I like this more, thank you!